src/main/java/li/h35/OraclePatchDownloader/OraclePatchDownloader.java

```diff
@@ -359,6 +405,20 @@ public static void main(String[] args) {
 
 			case 'p':
 				password = g.getOptarg();
+				if (password.matches("\\A[0-9A-Z_]+\\z")) {
```

Comment:
Think good idea, not sure about the match, \\A and \\z  , it's not ^ and $ for start and ending?

Reply:
Thanks for merging!

(`\A`, `\z`) match only at (beginning, end) of *string*, while (`^`, `$`) match also at (beginning, end) of *line*.  This can make a difference if the string that is being inspected can contain newlines.

For example, `/\Aabc\z/` would not match `"abc\n"`, but `/^abc$/` *would* match `"abc\n"`.

In practice this most often doesn't make a difference, and `^`, `$` is much easier to type, in particular on German keyboards :-).  I got used to `\A`, `\z`, in particular when inspecting user input.

Reply:
OK, what I have written above is true, but not relevant for the `String.matches` method, since that *always* matches against the whole string.  Which I tend to forget.

So the above call could be rewritten more simply and still correctly just as `password.matches("[0-9A-Z_]+")`.

Will implement that soon.

Reply:
`env:XXX` is also fine ... should I pack that into the next PR or would you like to do it separately?

Reply:
> `env:XXX` is also fine ... should I pack that into the next PR or would you like to do it separately?

you can pack it :-)